app/models/setting/provisioning.rb

```diff
@@ -28,7 +28,7 @@ def self.load_defaults
         self.set('libvirt_default_console_address', N_("The IP address that should be used for the console listen address when provisioning new virtual machines via Libvirt"), "0.0.0.0"),
         self.set('update_ip_from_built_request', N_("Foreman will update the host IP with the IP that made the built request"), false),
         self.set('use_shortname_for_vms', N_("Foreman will use the short hostname instead of the FQDN for creating new virtual machines"), false),
-        self.set('websockets_encrypt', N_("Should Foreman encrypt websockets (VNC console access). Choose on, off or auto."), "auto"),
+        self.set('websockets_encrypt', N_("Should Foreman encrypt websockets (VNC console access)"), false),
```

Comment:
Perhaps the value of SETTINGS[:require_ssl] would be a more accurate default?  (We use this for protocol selection in foreman_url already.)

If websockets_encrypt is disabled when SSL's enabled, then you get a mismatch and secure_headers prevents it loading.